### PR TITLE
feat: add query parameter to count query

### PR DESCRIPTION
**Pull request by aqibmohammadkhan**
## Issue

https://gravitee.atlassian.net/browse/APIM-10400

## Description
sample endpoint with query -> 
`http://localhost:8083/management/v2/environments/DEFAULT/apis/f8400543-759d-4b19-8005-43759d2b195f/analytics?type=COUNT&from=1752664013838&to=1753870126362&query=status:200`

Sample query -->

`{
  "size" : 0,
  "query" : {
    "bool" : {
      "must" : [ {
        "query_string" : {
          "query" : "status:200"
        }
      }, {
        "term" : {
          "api-id" : "f8400543-759d-4b19-8005-43759d2b195f"
        }
      }, {
        "range" : {
          "@timestamp" : {
            "from" : 1752664013838,
            "include_lower" : true,
            "to" : 1753870126362,
            "include_upper" : true
          }
        }
      } ]
    }
  }
}`

**without Query**. (Count = 24)
image

**With Query (Status:200)** (count 21, other 3 are statuscode 500)
image

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-lleyfvyvkh.chromatic.com)

### CODE REVIEW

File: gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/test/java/io/gravitee/repository/elasticsearch/v4/analytics/adapter/SearchRequestsCountByEventQueryAdapterTest.java [hunk 1]
```diff
@@ -38,7 +38,8 @@ public class SearchRequestsCountByEventQueryAdapterTest {
     void shouldAdaptQueryWithAllFields() throws Exception {
```
Comment (dev-marek) [inline gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/test/java/io/gravitee/repository/elasticsearch/v4/analytics/adapter/SearchRequestsCountByEventQueryAdapterTest.java:39]: Please add one test case to assert, that if query string is not empty, it is included in output query

File: gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/test/java/io/gravitee/repository/elasticsearch/v4/analytics/AnalyticsElasticsearchRepositoryTest.java [hunk 2]
```diff
@@ -969,7 +969,11 @@ void should_return_all_the_requests_count_by_entrypoint_for_a_given_api() {
             var to = now.plus(Duration.ofDays(1)).truncatedTo(ChronoUnit.DAYS);
             var result = cut.searchRequestsCountByEvent(
                 new QueryContext("org#1", "env#1"),
-                new RequestsCountByEventQuery(new SearchTermId(SearchTermId.SearchTerm.API, API_ID), new TimeRange(from, to))
+                new RequestsCountByEventQuery(
+                    new SearchTermId(SearchTermId.SearchTerm.API, API_ID),
+                    new TimeRange(from, to),
+                    Optional.empty()
```
Comment (dev-marek) [inline gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/test/java/io/gravitee/repository/elasticsearch/v4/analytics/AnalyticsElasticsearchRepositoryTest.java:975]: Please write an additional test with non-empty query string

**Review by dev-marek — CHANGES_REQUESTED** [review]
Only two small changes needed

File: gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/test/java/io/gravitee/repository/elasticsearch/v4/analytics/adapter/SearchRequestsCountByEventQueryAdapterTest.java (hunk 1 above)
Comment (aqibmohammadkhan) [reply gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/test/java/io/gravitee/repository/elasticsearch/v4/analytics/adapter/SearchRequestsCountByEventQueryAdapterTest.java:39]: done.

File: gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/test/java/io/gravitee/repository/elasticsearch/v4/analytics/AnalyticsElasticsearchRepositoryTest.java (hunk 2 above)
Comment (aqibmohammadkhan) [reply gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/test/java/io/gravitee/repository/elasticsearch/v4/analytics/AnalyticsElasticsearchRepositoryTest.java:975]: done

**Review by dev-marek — APPROVED** [review]
LGTM
